### PR TITLE
b-link component

### PR DESCRIPTION
**Pull request by tomeraronovsky**
b-link component

### CODE REVIEW

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div class="b-link">
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.component.html]: no need for wrapper

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+@import '../../style/colors';
+@import '../../style/typography';
+
+:host {
+  .b-link {
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss]: remove wrapper

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    component.config = { url: 'https://app.hibob.com', text: 'Learn more' };
+    component.color = LinkColor.primary;
+    component.openInNewWindow = true;
+    fixture.detectChanges();
+  });
+
+  it('link should be with the right url and text', () => {
+    const aElement = fixture.debugElement.query(By.css('a'));
+    expect(aElement.componentInstance.config.text).toEqual('Learn more');
+    expect(aElement.componentInstance.config.url).toEqual('https://app.hibob.com');
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.component.spec.ts]: cool

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import { LinkColor } from './link.enum';
+
+@Component({
+  selector: 'b-link',
+  templateUrl: './link.component.html',
+  styleUrls: ['./link.component.scss']
+})
+export class LinkComponent {
+  @Input() config: Link;
+  @Input() openInNewWindow: boolean;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.component.ts]: give default value

File: projects/ui-framework/src/lib/buttons-indicators/link/link.enum.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export enum LinkColor {
+  none = 'dark',
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.enum.ts]: none = 'none', no need for dark color, should be body color, also in scss

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.ts (hunk 4 above)
Comment (tomeraronovsky) [reply projects/ui-framework/src/lib/buttons-indicators/link/link.component.ts]: its false by default

File: projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+const note = `
+  ## Slider Element
+  #### Module
+  *SliderModule*
+
+  #### Properties
+  Name | Type | Description | Default value
+  --- | --- | --- | ---
+  config | object | link configuration - text, url |
+  openInNewWindow | boolean | open link in new window or not  | false (optional)
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts]: optional is on the param, not the default value
also need to implement the default in the controller

File: projects/ui-framework/src/public_api.ts [hunk 7]
```diff
@@ -67,6 +67,9 @@ export { ChipType } from './lib/buttons-indicators/chips/chips.enum';
 export { InfoStripModule } from './lib/buttons-indicators/info-strip/info-strip.module';
 export { InfoStripComponent } from './lib/buttons-indicators/info-strip/info-strip.component';
 export { Link } from './lib/buttons-indicators/info-strip/info-strip.types';
+// Link Module
+export { LinkModule } from './lib/buttons-indicators/link/link.module';
+export { LinkComponent } from './lib/buttons-indicators/link/link.component';
```
Comment (omrihecht) [inline projects/ui-framework/src/public_api.ts]: export interface as well

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss [hunk 8]
```diff
@@ -0,0 +1,16 @@
+@import '../../style/colors';
+@import '../../style/typography';
+
+:host {
+  a {
+    text-decoration-line: none;
+    line-height: 14px;
+    color: $grey-700;
```
Comment (omrihecht) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss]: does color: initial work?

File: projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+import { values } from 'lodash';
+const color = values(LinkColor);
+
+const linkStories = storiesOf(ComponentGroupType.ButtonsAndIndicators, module).addDecorator(
+  withKnobs
+);
+const template = `<b-link
+  [config]="config"
+  [openInNewWindow]="openInNewWindow"
+  [color]="color">
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts]: I think either all config should be in the config input, or everything in separate inputs.
not logical when url is in config object and the _blank in separate input...
@omrihecht

File: projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss (hunk 8 above)
Comment (galloween) [reply projects/ui-framework/src/lib/buttons-indicators/link/link.component.scss]: no. 'initial' will not work (will give default browser link color).
but 'unset' will (will give text color of parent).

File: projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts (hunk 9 above)
Comment (tomeraronovsky) [reply projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts]: I had interface that contain url & text so I wanted to use it here. @omrihecht WDYT ?

Comment (galloween) [reply projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts]: maybe the new interface should be extended with additional/optional properties?...
(then its backwards compatible)

export interface Link {
  text: string;
  url: string;
  openInNewWindow?: boolean;
  color?: string;
}

Comment (omrihecht) [reply projects/ui-framework/src/lib/buttons-indicators/link/link.stories.ts]: I agree, missed that, there should be one interface for the link @tomeraronovsky please update

**Pull request merged**